testing/cmake/test_export/CMakeLists.txt

```diff
@@ -8,11 +8,37 @@ add_executable(test_bench test_bench.cu)
 target_link_libraries(test_bench PRIVATE nvbench::main)
 enable_testing()
 add_test(NAME test_bench COMMAND "$<TARGET_FILE:test_bench>" --timeout 1)
+add_test(NAME nvbench_ctl COMMAND "$<TARGET_FILE:nvbench::ctl>")
+
+# Setup LD_LIBRARY_PATH for testing
+if (UNIX)
+  set(ctl_lib_path "")
+  set(cupti_lib_path "")
+
+  # Need to find installed libnvbench.so for installed nvbench-ctl.
+  # Not needed for build_tree test because of RUNPATH.
+  if (TEST_TYPE STREQUAL "INSTALL_TREE")
+    get_property(nvbench_config TARGET nvbench::nvbench
+      PROPERTY IMPORTED_CONFIGURATIONS
+    )
+    get_property(ctl_lib_path TARGET nvbench::nvbench
+      PROPERTY IMPORTED_LOCATION_${nvbench_config}
```

Comment:
@robertmaynard I need to redo how I detect `nvbench_config`, since this code only works when there's one `IMPORTED_CONFIGURATION` for `nvbench::nvbench`. What's the proper way to determine which configuration to use?

Reply:
For muti-config generators you don't have a great option.

You can iterate `IMPORTED_CONFIGURATIONS` to see all the built configurations for nvbench. But from there you will need to do something like this:
```
add_test(NAME mytest
         COMMAND ${CMAKE_COMMAND} -P <script> -DCONFIG=$<CONFIG>
                            -DTEST=$<TARGET_FILE:myexe>)
```

And have `script` set the LD_LIBRARY_PATH

Reply:
Thanks -- that sounds complicated 😅

Since we're only querying this to get the libdir for `LD_LIBRARY_PATH`, I'll just emit a warning and use the first one in the list. If it breaks for multiconfig later I'll fix it properly. I don't know that we'll actually run into the combination of UNIX + multiconfig builds for this to matter.